protocols/directfund/directfund.go

```diff
@@ -188,6 +188,12 @@ func (dfo *Objective) GetStatus() protocols.ObjectiveStatus {
 	return dfo.Status
 }
 
+// IsLedgerChannel returns true if the objective is direct funding a ledger channel.
```

Comment:
I think we'll need deeper changes to the direct fund protocol. At the moment, the protocol is geared toward funding a ledger channel. A ledger channel is a specific type of channel used to fund other (virtual) channels. We should probably refactor the direct fund protocol to remove the assumption that the protocol funds a ledger channel.

A ledger channel is also known as a consensus channel in the golang codebase.

Reply:
yes, but I can't pinpoint what makes a ledger channel different than a direct channel.
there is no flag at the state level.

Reply:
I removed this changes for now. But we definitely need to refactor ledger channels, so they will be a subtype of a direct funded channels

Reply:
> yes, but I can't pinpoint what makes a ledger channel different than a direct channel. there is no flag at the state level.

That is true, and we "get away with it" because we only have two types of channel currently (namely, Ledger Channels and Virtual Payment Channels).

 Currently the flow from API to store is:

`CreateLedgerChannel`[ API method ](https://github.com/statechannels/go-nitro/blob/2b0bc7c5998abe021970704f448d030844a05928/node/node.go#L244) -> [`directfund.Objective`](https://github.com/statechannels/go-nitro/blob/2b0bc7c5998abe021970704f448d030844a05928/protocols/directfund/directfund.go#L41) -> [`Channel`](https://github.com/statechannels/go-nitro/blob/8940b55c99f0aeb84bec9b486e5b53fa31bab03c/node/engine/store/store.go#L32). 

When the directfund objective completes, the channel is converted into a ConsensusChannel. When the ledger channel is closed, it is converted back into a regular `Channel`.

We may need to attach a new `type` property to a `Channel`. It can be an enum `Application, Ledger`.
When we call `CreateLedgerChannel` we get  `type=Ledger`, when we call `CreateDirectlyFundedApplicationChannel` we get `Application`.